includes/template-library/sources/local.php

```diff
@@ -1341,16 +1342,24 @@ public function print_blank_state_template( $current_type_label, $href, $descrip
 					?>
 				</h3>
 				<p><?php echo wp_kses_post( $description ); ?></p>
-				<a id="elementor-template-library-add-new" class="elementor-button elementor-button-success" href="<?php echo esc_url( $href ); ?>">
-					<?php
-					/* translators: %s: Template type label. */
-					printf( esc_html__( 'Add New %s', 'elementor' ), esc_html( $current_type_label ) );
-					?>
-				</a>
+				<?php if ( $this->is_license_valid() ) { ?>
+					<a id="elementor-template-library-add-new" class="elementor-button elementor-button-success" href="<?php echo esc_url( $href ); ?>">
+						<?php
+						/* translators: %s: Template type label. */
+						printf( esc_html__( 'Add New %s', 'elementor' ), esc_html( $current_type_label ) );
+						?>
+					</a>
+				<?php } else {
+					// TODO: Connect & Activate has to be here
+				} ?>
 			</div>
 		<?php
 	}
 
+	private function is_license_valid() {
+		return ( new License_Service )->is_valid();
```

Comment:
Why do you create a new instance of the License_Service? 
Haven't created a services class that registers all the services classes?